software-architecture/user-stories.qmd

```diff
@@ -0,0 +1,73 @@
+|User Access Administrator|see the individuals assigned to each role|so that I can easily audit who has access to which parts of the system|
+|User Access Administrator|see the groups/roles assigned to an individual|so that I can easily audit what kind of access a user has/has had|
+
+## Data Request Admin
+
+|As a... |I want to... |Because... |
+| --- | --- | --- |
+|Data Request Admin|see the applications submitted|so that I can check if they are completed correctly|
+|Data Request Admin|send an application back to the requester|so that they can fill out any missing details|
+|Data Request Admin|comment on a data request|so that I can inform either the requester or the data controller of my feedback|
+|Data Request Admin|forward a request to the data controller|so that they can make a decision on whether or not to grant access to the data|
+|Data Request Admin|put together a data extract script|so that I can provide the data requested|
+|Data Request Admin|pull the data out of the system|so that I can make the data available to the requestor|
+|Data Request Admin|see how many rows of data a specific request will generate|so that I can determine if the tables have been linked correctly|
+|Data Request Admin|select between formats for download of data|so that I can send the most appropriate format to the data requester|
+
+## Data Requester
+
+|As a... |I want to... |Because... |
+| --- | --- | --- |
+|Data Requester|submit a request for data|so that I can do new research or extend my current research|
+|Data Requester|see the progress of my request in the system|so that I know when I'm likely to hear of a decision|
+|Data Requester|see the variables that the system contains|so that I can decide if I should apply for data|
+|Data Requester|select the variables I need|so that I can put together a variable list for an application for data|
+|Data Requester|ask questions about variables|so that I can get clarificaton before asking for data|
+|Data Requester|select between different file formats when I submit a data request|so that I get the data in a format I can work with|
+
+## Data Contributor
+
+|As a... |I want to... |Because... |
+| --- | --- | --- |
+|Data Contributor|be able to create tables |so that I can store my data|
+|Data Contributor|be able to enter free text definitions for my variables|so that I can show the rest of the users what exactly has been captured|
+|Data Contributor|add fields to my tables|so that I can add more variables to the same research subject|
+|Data Contributor|add validation to my fields|so that the system will alert me, if there is data that doesn't fit in my upload|
+|Data Contributor|define a field (or set of fields) as a primary key for a table|so that I can link data between tables|
+|Data Contributor|define a field (or set of fields) as a foreign key for a table|so that I can link data between tables|
```

Comment:
I'm not sure the data contributors should be allowed to set any database keys.

Reply:
You are probably right, it should be one of the higher level admins who do this.  So far we don't have a specific admin for the data, I'm guessing that it would fall under the primary admin for now...